crv/crvBezierFields.cc

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2015 Scientific Computation Research Center
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+#include "crv.h"
+#include "crvBezier.h"
+#include "crvBezierShapes.h"
+#include "crvMath.h"
+#include "crvShape.h"
+#include "crvTables.h"
+#include "crvQuality.h"
+#include <apfField.h>
+#include <lionPrint.h>
+#include <cstdlib>
+#include <mth_def.h>
+#include <iostream>
+#include <pcu_util.h>
+
+namespace crv {
+
+static void convertVectorFieldInterpolationPoints(int n, int ne,
+    apf::NewArray<apf::Vector3>& nodes,
+    apf::NewArray<double>& c,
+    apf::NewArray<apf::Vector3>& newNodes){
+
+  for(int i = 0; i < ne; ++i)
+    newNodes[i].zero();
+
+  for( int i = 0; i < ne; ++i)
+    for( int j = 0; j < n; ++j)
+      newNodes[i] += nodes[j]*c[i*n+j];
+}
+
+static void convertScalarFieldInterpolationPoints(int n, int ne,
+    apf::NewArray<double>& nodes,
+    apf::NewArray<double>& c,
+    apf::NewArray<double>& newNodes)
+{
+
+  for(int i = 0; i < ne; ++i)
+    newNodes[i] = 0.;
+
+  for( int i = 0; i < ne; ++i)
+    for( int j = 0; j < n; ++j)
+      newNodes[i] += nodes[j]*c[i*n+j];
+}
+
+void convertInterpolationFieldPoints(apf::MeshEntity* e,
+    apf::Field* f,
+    int n, int ne, apf::NewArray<double>& c)
+{
+
+  apf::NewArray<apf::Vector3> l, b(ne);
+  apf::NewArray<double> ls, bs(ne);
+  apf::MeshElement* me =
+      apf::createMeshElement(f->getMesh(),e);
+  apf::Element* elem =
+      apf::createElement(f,me);
+
+  if (apf::getValueType(f) == apf::VECTOR) {
+    apf::getVectorNodes(elem,l);
+    convertVectorFieldInterpolationPoints(n, ne, l, c, b);
+    for(int i = 0; i < ne; ++i)
+      apf::setVector(f, e, i, b[i]);
+  }
+  else if (apf::getValueType(f) == apf::SCALAR) {
+    apf::getScalarNodes(elem, ls);
+    convertScalarFieldInterpolationPoints(n, ne, ls, c, bs);
+    for(int i = 0; i < ne; ++i)
+      apf::setScalar(f, e, i, bs[i]);
+  }
+  else
+    printf("Field type not implemented\n");
+  apf::destroyElement(elem);
+  apf::destroyMeshElement(me);
+}
+
+void convertInterpolatingFieldToBezier(apf::Mesh2* m_mesh, apf::Field* f)
+{
+  // TODO: to be completed
```

Comment:
@avinmoharana 

Is the TODO addressed here? If yes it has to be removed.

Also, we do not want to have commented code anywhere.

Reply:
@avinmoharana 

We do not want to have commented code in any case. If the todo is obsolete, please remove it.